xcbgen-rs/src/defs.rs

```diff
@@ -648,6 +1073,8 @@ impl Alignment {
     }
 
     /// Returns an alignment that is met by `self` and `other`.
+    // FIXME: `align = self.align.min(other.align)` strikes my as fishy. See my comment on
```

Comment:
`align`s are powers of two, so `min` will return the least restrictive alignment.

Reply:
Ah, yes. "Back then" I didn't know that this requires/assume `align` to be a power of two. I quickly grepped around in xcb-proto and didn't really find anything about this alignment business.

(And thought that this stuff matters for libxcb since C requires alignment memory access, but does not matter for x11rb since the generated code is perfectly fine with unaligned fields and does the right thing with them...)

Reply:
There is still some cases where we need to check the alignment.

If we have something like this:
```xml
<struct name="IHaveAPad4">
    <field type="CARD8" name="..."/>
    <pad align="4" />
    <field type="CARD32" name="..."/>
</struct>

<struct name="SomeStruct">
    <field type="CARD8" name="..."/>
    <field type="IHaveAPad4" name="..."/>
</struct>
```

Let's consider that we have a `SomeStruct` that starts at address zero, the `pad` of `IHaveAPad4` should have a length of 2. However, the parser of `IHaveAPad4` will calculate a pad length of 3, because it only sees its own `CARD8`. Thankfully, this situation never happens in the XML definitions. The align checker will enforce that, if a struct has a `<pad align="N">`, the beginning of the whole struct must be aligned at least to `N`.

Reply:
And the purpose of the `internal_align` field in `ComplexAlignment` is to track the `<pad align= >`s inside a struct.

Reply:
Ah, thanks. I came across (part of) the code for this and was quite confused / did not understand what was going on. This now makes perfect sense.